lua/lualine/components/diagnostics.lua

```diff
@@ -0,0 +1,41 @@
+-- Copyright (c) 2020-2021 hoob3rt
+-- MIT license, see LICENSE for more details.
+
+local function diagnostics(options)
+  return function()
+    local error_count, warning_count, info_count = 0,0,0
+    if options.sources~=nil then
+      for _, source in ipairs(options.sources) do
+        if source == 'lsp' then
+          error_count = error_count +  vim.lsp.diagnostic.get_count(0, 'Error')
+          warning_count = warning_count +  vim.lsp.diagnostic.get_count(0, 'Warning')
+          -- info_count = info_count +  vim.lsp.diagnostic.get_count(0, 'Info')
+        end
+      end
+    end print(error_count)
+    local result = {}
+    local symbols = {
+      'E',
```

Comment:
How about some nice icons too?

Reply:
from https://www.nerdfonts.com/cheat-sheet:
error -> nf-mdi-close_circle_outline (f659)
warning -> nf-mdi-alert_outline (f529)
info/hint -> nf-mdi-information_outline (f7fc)

Also, colouring is important - preferably: errors red; warnings yellow; info/hint white.

Reply:
Yep, coloring will be similar to #61, decent out of the box with further customization available